phpBB/phpbb/extension/extension_interface.php

```diff
@@ -27,6 +27,13 @@
 	public function is_enableable();
 
 	/**
+	* Indicate why the extension can not be enabled.
+	*
+	* @return string|array with actual processed language string(s) (not language keys), or empty
+	*/
+	public function not_enableable_reason();
```

Comment:
No need for complex parsing, that's what we are already supporting :
 - $language->lang_array('string', array)
 - $language->lang('string')

For the language string not loaded because the extension is not available I see the issue. But we could add in the core a bunch of messages corresponding to the most frequent issue (wrong version of php, wrong version of phpbb, missing php ext, missing dependent ext etc...)

- `''` would just be an empty message, that's not our problem as it won't cause any issue.
- `[]` should use the same default than null

About the truish / falsish values I would ignore that. PHP is a complex language with nuance. The weak typing is a feature but we are not forced to use it everywhere and for everything. It will just make things more complex.

A third solution could be to support and handle exceptions: if an exception cannot be enabled, it just throw an exception in a pre_enable() method per example.

Ref: https://github.com/phpbb/phpbb/blob/master/phpBB/phpbb/language/language.php